cachi2/core/package_managers/yarn/project.py

```diff
@@ -91,11 +97,21 @@ def package_manager(self) -> Optional[str]:
         """Get the package manager string."""
         return NotImplemented
 
+    @package_manager.setter
+    def package_manager(self, package_manager: str) -> None:
+        """Set the package manager string."""
+        self._data["packageManager"] = package_manager
+
     @classmethod
     def from_file(cls, file_path: RootedPath) -> "PackageJson":
         """Parse the content of a package.json file."""
         return NotImplemented
 
+    def to_file(self) -> None:
```

Comment:
Would it make sense to call `yarn set version` instead of writing the file manually?

Reply:
I think `yarn set version` does a lot more than what we want, like downloading the `yarn` release that you specify

Reply:
If it downloads the yarn release, would it be able to replace Corepack?

Reply:
It just saves the `.cjs` file in`yarnPath` AFAIK (but proably does other things as well, this was just an example)